src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/locations/CreateLocationRequest.kt

```diff
@@ -0,0 +1,88 @@
+package uk.gov.justice.digital.hmpps.nomisprisonerapi.locations
+
+import com.fasterxml.jackson.annotation.JsonInclude
+import io.swagger.v3.oas.annotations.media.Schema
+import jakarta.validation.constraints.Size
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyInternalLocation
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AgencyLocation
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.HousingUnitType
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.InternalLocationType
+
+@JsonInclude(JsonInclude.Include.NON_NULL)
+@Schema(description = "Location creation request")
+data class CreateLocationRequest(
+
+  @Schema(description = "Whether certified for use", required = false, example = "true", defaultValue = "false")
+  val certified: Boolean = false,
+
+  @Schema(
+    description = "Whether a CELL, VISIT room, Kitchen etc (Ref type ILOC_TYPE)",
+    required = true,
+    example = "LAND",
+  )
+  val locationType: String,
```

Comment:
might be worth having a full allow list - will help with the generated client code being able to use a enum when the classes are generated using the OpenAPI gradle plugin

Reply:
Done